### PR TITLE
Flagify animal empathy/discord traits

### PR DESCRIPTION
**Pull request by Standing-Storm**
#### Summary
Features "Flagify animal empathy/discord traits"

#### Purpose of change

More flexibility, less hardcoded, more extensibility. 

#### Describe the solution

Change hardcoding of animal traits from checking for the specific trait to checking for a flag. Add the flag to each basic animal trait. Check the Discord flags first, allowing for curses, portal storm effects, or other temporary ways to make animals hate you.

Remove hardcoding of animal reactions from Nature's Communion spell in Magiclysm, use the flag instead.

Remove the errant threshold requirement in Magiclysm from Animal Kinship (which prevents anyone else from gaining that mutation). 

#### Testing

A little hard to test because the effects are not that dramatic (e.g. Animal Discord does not actually make animals hate you--it makes animals that already are attacking you attack you for longer), but nothing seems obviously wrong.

### CODE REVIEW

**Comment by GuardianDll** [comment]
won't it be better to move this entire thing to enchantments?

**Comment by Standing-Storm** [comment]
Enchantments would need to have some formula defined to calculate from--right now the formula for the traits isn't linear (ANIMALDISCORD reduces monster morale by 5 for both the initial and upgraded flags but ANIMALEMPATHY increases it by 55/80), so it's not a simple matter of converting the existing bonuses straight to enchants.

**Comment by Standing-Storm** [comment]
If you'd like to handle that after you return from your vacation, I'll gladly defer to you.

**Comment by GuardianDll** [comment]
I'll do it then

**Comment by GuardianDll** [comment]
You know what, after a big digging i refrain from what i say
It should be jsonified, and maybe even using enchantment framework, but not as another add/multiply value, but as it's own thing, that i am sadly not sure how to pull
I am not sure it is really worth moving it to values?

**Comment by Standing-Storm** [comment]
> I am not sure it is really worth moving it to values?

I can think of some things I would do it with it if it were adjustable--have the bonus scale with spell level, you could have a high level druid spell where animals were eating out of your hand, let feral druids curse you so that animals would attack you on sight, some kind of Magiclysm rare celestial event that makes all animals aggressive, etc. 

Not sure if that justifies the work it would take.

**Comment by fairyarmadillo** [comment]
> yes that's what i mean, one enchantment for animal morale, and another for animal anger

Chiming in to say that in my experience, it is incredibly difficult to get arbitrary adjustments to enemy morale and aggression to work out in any kind of predictable way, and that this would be many times worse if you were applying a scalable value to more than one type of enemy. If you look at, for example, black bear, wild boar, moose, and coyote, their numerical values are _all over the place_. This is because, as far as I can tell, they weren't designed according to one universal metric, but rather to get each critter to work properly with its aggression and fear triggers.

The traits currently don't work very well, but it's sort of fine because they're pretty small and flat values so it doesn't matter that much. I think you would find worse results with larger or more variable values, which would both make it clear to players that the whole thing wasn't working properly and make it difficult for anyone working on enchantments to do anything productive with them.

TLDR: It should probably be an enchantment but I want to warn you that it's already sort of broken and would get way worse if you poked it. Fixing it would probably require some pretty drastic measures vis a vis the morale/aggression system itself.

**Comment by Maleclypse** [comment]
Clang angry
